cpp/modmesh/transform/fourier.hpp

```diff
@@ -3,6 +3,10 @@
 #include <modmesh/math/math.hpp>
 #include <modmesh/buffer/buffer.hpp>
 
+#if defined(BUILD_CUDA)
```

Comment:
I prefer to avoid any `BUILD_CUDA` macro from outside `device/cuda`.  How much work it takes to make it?